From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 30 Nov 2010 19:59:51 -0500
Subject: [scsi] qla4xxx: cleanup qla4xxx_wait_for_hba_online
Message-id: <20101130195951.10450.72780.sendpatchset@localhost.localdomain>
Patchwork-id: 29702
O-Subject: [RHEL 5.6 PATCH 11/28] qla4xxx: cleanup qla4xxx_wait_for_hba_online
Bugzilla: 656999

Bugzilla
--------

Bug 656999 (https://bugzilla.redhat.com/show_bug.cgi?id=656999)

Upstream Status
---------------

RHEL 5 specific.

Description
-----------

>From 5bb4cd0dd5c20621ecfcc2d770f0df32d5e2ccad Mon Sep 17 00:00:00 2001
From: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Date: Sat, 20 Nov 2010 14:29:34 -0800
Subject: [PATCH 12/29] qla4xxx: cleanup qla4xxx_wait_for_hba_online

Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Signed-off-by: Prasanna Mumbai <prasanna.mumbai@qlogic.com>

diff --git a/drivers/scsi/qla4xxx/ql4_os.c b/drivers/scsi/qla4xxx/ql4_os.c
index 449b550..ad2ae6e 100644
--- a/drivers/scsi/qla4xxx/ql4_os.c
+++ b/drivers/scsi/qla4xxx/ql4_os.c
@@ -2183,10 +2183,8 @@ static int qla4xxx_wait_for_hba_online(struct scsi_qla_host *ha)
 	wait_online = jiffies + (60 * HZ);
 	while (time_before(jiffies, wait_online)) {
 
-		if (adapter_up(ha))
+		if (test_bit(AF_ONLINE, &ha->flags) != 0)
 			return QLA_SUCCESS;
-		else if (ha->retry_reset_ha_cnt == 0)
-			return QLA_ERROR;
 
 		msleep(2000);
 	}
